pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix

```diff
@@ -0,0 +1,155 @@
+  dontWrapGApps = !integrateWithSystem;
+  dontWrapQtApps = !integrateWithSystem;
+
+  nativeBuildInputs = [ pkgconfig pythonPackages.gyp cmake ninja gcc9 ]
+    ++ optionals (!integrateWithSystem) [ wrapGAppsHook wrapQtAppsHook ];
+
+  buildInputs = [
+    qtbase qtimageformats ffmpeg_4 openalSoft lzma lz4 xxHash
+    zlib minizip openssl libtgvoip rlottie-tdesktop range-v3
+  ] ++ optionals (!integrateWithSystem) [ gtk3 libappindicator-gtk3 ];
+
+  propagatedBuildInputs = optional withFont open-sans;
+
+  enableParallelBuilding = true;
+
+  NIX_CFLAGS_COMPILE = [
+    "-I${getDev libtgvoip}/include/tgvoip"
+    # See Telegram/gyp/qt.gypi
+    "-I${getDev qtbase}/mkspecs/linux-g++"
+  ] ++ concatMap (x: [
+    "-I${getDev qtbase}/include/${x}"
+  ]) [ "QtCore" "QtGui" "QtDBus" ] ++ concatMap (x: [
+    "-I${getDev qtbase}/include/${x}/${qtbase.version}"
+    "-I${getDev qtbase}/include/${x}/${qtbase.version}/${x}"
+  ]) [ "QtCore" "QtGui" ];
+  CPPFLAGS = NIX_CFLAGS_COMPILE;
+
+  preConfigure = ''
+    gyp \
+      -Dapi_id=17349 \
+      -Dapi_hash=344583e45741c457fe1862106095a5eb \
+      -Dbuild_defines=${GYP_DEFINES} \
+      -Dlottie_use_cache=1 \
+      -Gconfig=Release \
+      --depth=Telegram/gyp \
+      --generator-output=../.. \
+      -Goutput_dir=out \
+      --format=cmake \
+      Telegram/gyp/Telegram.gyp
+
+    cd out/Release
+
+    NUM=$((`wc -l < CMakeLists.txt` - 2))
+    sed -i "$NUM r ${./CMakeLists.inj}" CMakeLists.txt
+  '';
+
+  installPhase = ''
+    install -Dm755 Telegram $out/bin/kotatogram-desktop
+
+    mkdir -p $out/share/applications $out/share/kservices5
+    install -m444 "$src/lib/xdg/kotatogramdesktop.desktop" "$out/share/applications/kotatogramdesktop.desktop"
+    sed "s,/usr/bin,$out/bin,g" $src/lib/xdg/tg.protocol > $out/share/kservices5/tg.protocol
+    for icon_size in 16 32 48 64 128 256 512; do
+      install -Dm644 "../../../Telegram/Resources/art/icon''${icon_size}.png" "$out/share/icons/hicolor/''${icon_size}x''${icon_size}/apps/kotatogram.png"
+    done
+  '' + optionalString integrateWithSystem ''
+    mkdir -p $out/share/KotatogramDesktop/autostart
+    install -m444 "${./autostart.desktop}" "$out/share/KotatogramDesktop/autostart/kotatogramdesktop.desktop"
+  '';
+
+  postFixup = optionalString (!integrateWithSystem) ''
+    # This is necessary to run Kotatogram in a pure environment.
+    # We also use gappsWrapperArgs from wrapGAppsHook.
+    wrapProgram $out/bin/kotatogram-desktop \
+      "''${gappsWrapperArgs[@]}" \
+      "''${qtWrapperArgs[@]}" \
+      --prefix PATH : ${xdg_utils}/bin \
+      --set XDG_RUNTIME_DIR "XDG-RUNTIME-DIR"
+    sed -i $out/bin/kotatogram-desktop \
+      -e "s,'XDG-RUNTIME-DIR',\"\''${XDG_RUNTIME_DIR:-/run/user/\$(id --user)}\","
+  '';
+
+  meta = {
+    description = "Kotatogram – experimental Telegram Desktop fork";
+    longDescription = ''
+      Unofficial desktop client for the Telegram messenger, based on Telegram Desktop.
```

Comment:
Would be helpful for users to add something like this:
> It contains some useful (or purely cosmetic) features, but they could be unstable. A detailed list is available here: https://kotatogram.github.io/changes

Reply:
Fixed